File: app/src/main/java/com/stevenfrew/beatprompter/midi/ByteValue.java
package com.stevenfrew.beatprompter.midi;

public abstract class ByteValue extends Value {
    byte mValue;
    ByteValue(byte value)
    {
        mValue=value;
    }

    @Override
    byte resolve(byte[] arguments, byte channel) {
        return mValue;
    }

    @Override
    boolean matches(Value otherValue) {
        if(otherValue instanceof ByteValue)
            return ((ByteValue) otherValue).mValue==mValue;
        return otherValue instanceof WildcardValue;
    }

    @Override
    public String toString()
    {
        return ""+mValue;
    }
}

File: app/src/main/kotlin/com/stevenfrew/beatprompter/midi/ByteValue.kt
package com.stevenfrew.beatprompter.midi

abstract class ByteValue internal constructor(internal var mValue: Byte) : Value() {

    override fun resolve(arguments: ByteArray, channel: Byte): Byte {
        return mValue
    }

    override fun matches(otherValue: Value?): Boolean {
        return if (otherValue is ByteValue) otherValue.mValue == mValue else otherValue is WildcardValue
    }

    override fun toString(): String {
        return "" + mValue
    }
}

